README.md

```diff
@@ -50,26 +61,41 @@ The `GameBoard` class is the core of the game logic and user interaction. It con
 - **`__init__(self, size)`**: Initializes the game board with a specified size and sets up the initial grid with empty cells.
 
 - **`initialize_empty_coordinates`**: Go troght all the grid as save the empty spaces in a set()
+   - Time Complexity: O(n^2) - where n is the size of the grid. This method iterates through each cell of the grid.
+
 
 - **`update(self)`**: Prints the current state of the game grid to the terminal.
+  - Time Complexity: O(n^2) - as it iterates through each cell for printing.
 
 - **`read_user_input(self)`**: Listens for arrow key inputs using the `keyboard` library to determine the user's desired move direction.
```

Comment:
Need to be updated... check the code, becaus enow the possible keys are inside a dictionary with also possibility to press 'r' -> ranking
'u' -> undo
and maybe more... check it out and update i

Reply:
direction_keys = {'left': 'left arrow', 'right': 'right arrow', 'up': 'up arrow', 'down': 'down arrow', 'undo': 'u', 'ranking': 'r', 'quit': 'q'}

Reply:
direction_keys = {'left': 'left arrow', 'right': 'right arrow', 'up': 'up arrow', 'down': 'down arrow', 'undo': 'u', 'ranking': 'r', 'quit': 'q'}